client/question/services/FeedbackGeneratorService.js

```diff
@@ -268,28 +360,45 @@ tie.factory('FeedbackGeneratorService', [
      * @private
      */
     var _getCorrectnessTestFeedback = function(
-      outputFunctionName, testCase, observedOutput) {
+      outputFunctionName, testCase, observedOutput, testSuiteId,
+      testSuiteIndex) {
       var allowedOutputExample = testCase.getAnyAllowedOutput();
       var feedback = FeedbackObjectFactory.create(
         FEEDBACK_CATEGORIES.INCORRECT_OUTPUT_FAILURE);
-      // TODO(eyurko): Add varied statements for when code is incorrect.
-      feedback.appendTextParagraph('Your code produced the following result:');
-      if (outputFunctionName) {
-        feedback.appendCodeParagraph(
-          'Input: ' + _jsToHumanReadable(testCase.getInput()) + '\n' +
-          ('Output (after running ' + outputFunctionName + '): ' +
-            _jsToHumanReadable(observedOutput)));
-        feedback.appendTextParagraph(
-          'However, the expected output of ' + outputFunctionName + ' is:');
+      if (testSuiteIndex == lastTestSuiteIndex) {
+        // Same test suite; attempt number > 1
+        attemptCounter[testSuiteId] += 1;
+        if (attemptCounter[testSuiteId] == 2 && testSuiteId != 'SAMPLE_INPUT') {
+          // Display expected output
+          feedback.appendTextParagraph(
```

Comment:
Minor style nit: when a list of multiple args spans more than one line, break after the first '('. So, here, you could do:

              feedback.appendTextParagraph(_selectCorrectnessFeedback(
                correctnessFeedbackText, 'expected_output'));

Reply:
done